hologram_dashboard/fetcher/hologram_data_source_availability_fetcher.h

```diff
@@ -27,32 +24,32 @@
 #include "fetcher/proto/hologram_availability.pb.h"
 #include "fetcher/proto/hologram_config.pb.h"
 #include "hologram_data_fetcher.h"
-
-DEFINE_string(chipper_batch_job_cell, "", 
-    "The job running cell of Chipper Batch.");
-DEFINE_string(chipper_gdpr_batch_job_cell, "", 
-    "The job running cell of Chipper GDPR pipeline.");
-DEFINE_string(config_file_path, "", 
-    "Specifies where the config file can be located.");
+#include "flags.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/btree_map.h"
 
 namespace wireless_android_play_analytics {
 
 class HologramDataSourceAvailabilityFetcher: public HologramDataFetcher {
 
 public:
+    // Populates the system_to_cell_map_ with flag input
+    HologramDataSourceAvailabilityFetcher();
     void Process() override;
 
 private:
+    FRIEND_TEST(FetcherTest, ConstructorMissingFlags);
+    FRIEND_TEST(FetcherTest, ConstructorValidFlags);
     FRIEND_TEST(FetcherTest, InvalidAcquireConfig);
     FRIEND_TEST(FetcherTest, ValidAcquireConfig);
     // Poulates hologram_config_ and ends the program if the path provided leads
     // to wrong file or malformed file.
     void AcquireConfig(const std::string& config_file_path);
     
-    std::unordered_map<std::string, std::string> system_to_cell_map_;
+    absl::flat_hash_map<System, std::string> system_to_cell_map_;
```

Comment:
system_to_job_running_cell_

I don't put "map" as suffix, as it is already clear by variable type. But you can make the call.

Reply:
ya that makes sense i removed it